exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,305 @@
+                                         [&](const tfc::ipc_ruler::signal& signal) {
+                                           return std::ranges::any_of(banned_signals_.begin(), banned_signals_.end(),
+                                                                      [&](const std::string& banned_string) {
+                                                                        return signal.name.find(banned_string) !=
+                                                                               std::string::npos;
+                                                                      });
+                                         }),
+                          active_signals_.end());
+  }
+
+  // This function checks the type of the signal in order to determine how to read from the slot
+  auto handle_signal(tfc::ipc_ruler::signal& signal) -> void {
+    switch (signal.type) {
+      case details::type_e::_bool: {
+        run_slot<tfc::ipc::details::type_bool, bool>(signal);
+        break;
+      }
+      case details::type_e::_int64_t: {
+        run_slot<tfc::ipc::details::type_int, int>(signal);
+        break;
+      }
+      case details::type_e::_uint64_t: {
+        run_slot<tfc::ipc::details::type_uint, uint>(signal);
+        break;
+      }
+      case details::type_e::_double_t: {
+        run_slot<tfc::ipc::details::type_double, double>(signal);
+        break;
+      }
+      case details::type_e::_string: {
+        run_slot<tfc::ipc::details::type_string, std::string>(signal);
+        break;
+      }
+      case details::type_e::_json: {
+        run_slot<tfc::ipc::details::type_json, std::string>(signal);
+        break;
+      }
+      case details::type_e::unknown: {
+        break;
+      }
+    }
+  }
+
+  // This function runs the coroutine for the slot
+  template <typename slot_type, typename value_type>
+  auto run_slot(tfc::ipc_ruler::signal& signal) -> void {
+    slots_.push_back(std::make_shared<details::slot<slot_type>>(ctx_, signal.name));
+    auto& slot = std::get<std::shared_ptr<details::slot<slot_type>>>(slots_.back());
+    slot->connect(signal.name);
+    asio::co_spawn(mqtt_client_->strand(), slot_coroutine<slot_type, value_type>(*slot, signal.name), asio::detached);
+  }
+
+  // This function stops the coroutines that are currently running and then restarts them
+  auto restart() -> void {
+    stop_coroutine_ = true;
+    ctx_.run_for(std::chrono::milliseconds(20));
+    logger_.info("Signal list has been updated, reloading all signals");
+    load_signals();
+    stop_coroutine_ = false;
+  }
+
+  // This function is called when a signal is received
+  void update_signals([[maybe_unused]] sdbusplus::message::message& msg) noexcept { restart(); }
+
+  // This function is called when a value is received and its value needs to be converted to a string
+  template <typename value_type>
+  auto convert_to_string(value_type value) -> std::string {
```

Comment:
how about using `fmt::format("{}", value)` instead of this function?

Reply:
that's a better solution, deleting this function.